### DISCUSSION TITLE
Workflow Question - Submit posts using a different system

### DISCUSSION THREAD

**Discussion by raramuridesign** [GetPublii/Publii#1478] [General]
Hi 
We are considering using Publii for content generated by our users. Currently this is all done on a CMS. 
The proposed idea we have is as follows.

1. We would continue to receive the posts / articles using the current CMS
2. Export the content using an export tool (we can customise as we require)
3. Then import into Publii in whatever format we need

What we would like to know is the following:
1. How would we handle images when importing? ie featured images
2. How are authors new and old created so the author pages are shown?
3. Can Author profiles be extended with more information?
4. Is there anything else we should be aware of which could potentially be an issue? 

If this approach sounds too cumbersome and does not look like a solid solution, then is there another approach which could be recommended ?

Looking forward to the feedback.
Matthew

**Comment by internettips** [comment]
@raramuridesign 

Your questions are probably best answered by the developers @dziudek and / or @bobmitro , however, as I've been using the superb Publii for some years, maybe some additional feedback might help too.

You said:

> Hi We are considering using Publii for content generated by our users. Currently this is all done on a CMS. The proposed idea we have is as follows.
> 
> 1. We would continue to receive the posts / articles using the current CMS
> 2. Export the content using an export tool (we can customise as we require)
> 3. Then import into Publii in whatever format we need

If your current CMS is WordPress, then Publii can import an entire WordPress website. Details at: https://getpublii.com/docs/import-wordpress-into-static-html-site.html

If you just want to 'import' individual posts from multiple authors, there are probably various ways to do this. Perhaps the simplest and technically the safest option of course is just to get the text from each author ideally in plain text format, then create a new page in your Publii site, paste the text into the page and format it properly as required using the various tools and commands Publii provides. While somewhat tedious, this method is more likely to ensure that your final post / page is more SEO friendly, etc.

If you have another method in mind that interacts directly with the Publii .sqlite database, then that route is more ... technically challenging and potentially damaging to your site if get something wrong. If you know what you're doing, this route might work for you, though with a big caution :-)

One key point: I recommend that you use the built-in Publii tool to back up your site often and especially before making potentially dramatic code changes 'outside' of Publii.

For the other questions, you said:

> What we would like to know is the following:
> 
> 1. How would we handle images when importing? ie featured images

If you use a well designed and well structured Publii theme, such as those in the Publii Premium themes at: https://marketplace.getpublii.com/themes/ , I've found that so long as your source image is somewhere near a size that would typically display on say a typical PC or Apple MacBook, the Publii software will take that source image and work out all the other relevant image sizes needed to populate the various types of web pages in the theme. So for example, in the Art Gallery Publii Premium theme, a typical source image might be something like 1920  ×  1280 pixels. However, if your source image is larger, Publii will automatically size it down (or up) to ensure the right size image is in place for whatever part of the default is your using.

If you import an entire WordPress website, Publii will import the existing images within your WordPress source pages. If you use another CMS, such as Joomla, you can export the site from Joomla to WordPress format, then import the WordPress file into Publii. 

> 2. How are authors new and old created so the author pages are shown?

Your website theme in Publii will have one default author. For additional authors, you just use the relevant command within Publii and your Publii theme to create additional authors. Again, if you use a Premium Publii theme, probably more of the SEO work is done for you in relation to authors. However, having said that, often the free Publii themes at: https://marketplace.getpublii.com/themes/ can be remarkably impressive.

> 3. Can Author profiles be extended with more information?

The short answer is yes. Though typically, the default Publii themes tend to provide an information box for brief author profiles. If you want to have a full page about an author, one option is you could simple create a new content page in Publii, and link to that page from the relevant Publii author profile box.

If you wanted something more complex, you can modify Publii themes further, but you would then need to ensure that your changes stay in place when theme updates and Publii CMS updates become available. Details at: https://getpublii.com/dev/theme-overrides/ and for full developer information at: https://getpublii.com/dev/

You also said:

> 4. Is there anything else we should be aware of which could potentially be an issue?

Publii is mostly designed for single users, though you can have multiple people working on a Publii website (https://getpublii.com/docs/publii-on-multiple-computers-via-dropbox.html) -- you just need to co-ordinate between users. 

I don't think Publii is designed to allow multiple authors to interact with their content on a single Publii website: you, as the site owner, or one or two others who have full access to the entire website via Dropbox, could likewise work on the site. But naturally, you wouldn't want to give multiple authors access to your entire website, so they would need to send you their content for you to include in your site.

You also said:

> 
> If this approach sounds too cumbersome and does not look like a solid solution, then is there another approach which could be recommended ?
> 
> Looking forward to the feedback. Matthew

Much depends on what your requirements are. For me personally, I can easily see how you could import (say) an entire WordPress site into Publii, then 'accept' various author submissions ongoing. But you would need to create and manage the author pages.

One of the really fantastic things about Publii sites is that when they are created and structured using just the Publii default settings and options available in the Premium themes (and maybe some or most free themes), much of the SEO gets translated in the right format for you. In addition, probably all of the Publii Premium themes have various ARIA codes for better accessibility built in too. In addition, you can install various Publii plugins too that will enhance your final website even further. So lots of options to think about.

Given the seemingly increasing risk with most CMSes that store their content databases online, Publii is a refreshing solution that doesn't do that, so to me at least, seems excellent.

Hope all this helps.

Maybe others can suggest more options and ideas.

**Reply by raramuridesign** [reply]
@internettips Thank you for the detailed answer. 

I am familiar with publii and all the different functions and have been using it for sometime. However the last time I did an import of content was sometime ago.

I was hoping that there would be different import options, at the moment its only WP and it would be useful to have CSV with a full example of all the data we could potentially add. That would be really useful as it would then make the import job a lot smoother. 

I would prefer to have a very simple export / import process. That in a real world is a nice to have idea!

My initial feeling right now, is to ensure content is imported and all images are refereced to a CDN domain. 

Perhaps in a small way we could even get a CSV version. I will however explore the SQL light DB as an option.

:-)
M

**Reply by candidexmedia** [reply]
> I was hoping that there would be different import options, at the moment its only WP and it would be useful to have CSV with a full example of all the data we could potentially add. That would be really useful as it would then make the import job a lot smoother.

> Perhaps in a small way we could even get a CSV version. I will however explore the SQL light DB as an option.

Definitely look into the structure of the db.sqlite (using a test site to start). I've been able to open, view, and make simple edits to the db using [SqliteStudio](https://sqlitestudio.pl/) and [DB Browser for SQLite](https://sqlitebrowser.org/). If I remember correctly, both apps are able to import/export in csv.

Keep in mind that posts are stored based on the Editor selected. So, WYSIWYG posts are store as HTML (if I remember correctly), and I believe Markdown would be in Markdown. That is something to keep in mind when bringing posts from one CMS to another.

> Publii is mostly designed for single users

I echo what @internettips mentioned here. Depending on the volume of posts and number of contributors for your site, Publii may or may not be the right fit... Something like Ghost CMS (which can be quickly self-hosted on apps like [Railway](https://railway.app/templates?category=Blogs) and [Pikapods](https://www.pikapods.com/pods?run=ghost)) may cut down on the conversion between CMSs by giving all contributors access, but with different [user roles / restrictions](https://ghost.org/help/managing-your-team/). [It's what FreeCodeCamp uses](https://www.freecodecamp.org/news/developer-news-style-guide/) to allow their many contributors to write posts directly inside the app, but not hit publish until the content is vetted.

**Comment by raramuridesign** [comment]
@candidexmedia Thank you for the information. I am going to review these options and see what the best way forward is. 
Thanks
M
